sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -982,6 +982,28 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     spark.sessionState.conf.clear()
   }
 
+  test("SPARK-19218 Fix SET command to show a result correctly and in a sorted order") {
+    val overrideConfs = sql("SET").collect()
+    sql(s"SET test.key3=1")
+    sql(s"SET test.key2=2")
+    sql(s"SET test.key1=3")
+    val result = sql("SET").collect()
+    assert(result ===
+      (overrideConfs ++ Seq(
+        Row("test.key1", "3"),
+        Row("test.key2", "2"),
+        Row("test.key3", "1"))).sortBy(_.getString(0))
+    )
+
+    // Previsouly, `SET -v` fails with NPE during decoding for null value.
+    import SQLConf._
+    SQLConfigBuilder("spark.test").doc("doc").stringConf.createWithDefault(null)
+
+    val result2 = sql("SET -v").collect()
+    assert(result2 === result2.sortBy(_.getString(0)))
+    spark.sessionState.conf.clear()
```

Comment:
This will not drop the `spark.test`. We need to introduce a function for testing only
```Scala
  // For testing only
  private[sql] def unregister(entry: ConfigEntry[_]): Unit = sqlConfEntries.synchronized {
    sqlConfEntries.remove(entry.key)
  }
```

Create a separate test case; then call the following code in a finally block: `SQLConf.unregister(confEntry)`

Will be back after a few hours.